frontend/server/src/Controllers/Run.php

```diff
@@ -533,9 +533,13 @@ public static function apiCreate(\OmegaUp\Request $r): array {
             self::$log->error('Call to \OmegaUp\Grader::grade() failed', $e);
             throw $e;
         }
+        $userId = null;
```

Comment:
qué pasa si aquí se usa

```suggestion
        $userId = $r->identity->user_id;
```

? o más bien, por qué `$r->identity->user_id` ya no se considera en ningún lugar?

Reply:
Tienes razón: siempre debe prevalecer el `user_id` de la identidad seleccionada. En caso de no encontrar nada ahí entonces si se puede ir a buscar a la identidad logueada.